apple/internal/apple_framework_import.bzl

```diff
@@ -245,6 +265,14 @@ def _apple_static_framework_import_impl(ctx):
     if imported_swiftmodules:
         swift_toolchain = ctx.attr._toolchain[SwiftToolchainInfo]
         swift_linkopts = swift_common.swift_runtime_linkopts(True, swift_toolchain)
+        if ctx.var["COMPILATION_MODE"] == "dbg":
+            cpu = ctx.fragments.apple.single_arch_cpu
+            ast_files = _swiftmodules_for_cpu(imported_swiftmodules, cpu)
+            objc_provider_fields["link_inputs"] = depset(ast_files)
+            swift_linkopts.extend([
```

Comment:
@allevato is there a way to reuse something in rules_swift to propagate the swift modules and linkopts without repeating the linker flags here?

Reply:
Not with the APIs that exist today. The catch is that the way module ASTs are handled for debugging is platform-specific; on Apple platforms the .swiftmodule files are passed to the linker with `-add_ast_path`, but on Linux, a separate `swift -modulewrap` action is run that embeds the module in a `.o` file and that gets passed to the linker. When using the _compilation_ API this is all handled in rules_swift for you, but we don't provide hooks to take an already precompiled module and do the right thing.

It's definitely not a bad idea to expose a standalone API to generalize this (the `swift_import` rule would benefit from it), but it's extremely low-priority.

Reply:
> `swift-modulewrap` action is run that embeds the module in a `.o` file and that gets passed to the linker

TIL!